aten/src/ATen/native/quantized/cpu/qlinear.cpp

```diff
@@ -976,14 +1105,22 @@ static at::Tensor linear_int8_with_onednn_weight(
       );
     }
     if (binary_post_op == "sum") {
-      auto expected_dtype = output_dtype.has_value() ? output_dtype.value() : c10::kByte;
+      auto expected_dtype = output_dtype.has_value() ? output_dtype.value() : input.scalar_type();
       TORCH_CHECK(
           other.value().scalar_type() == expected_dtype,
           "onednn qlinear: the dtype of extra input for binary post op should be ", expected_dtype,
           " (same as output dtype), but got ", other.value().scalar_type()
       );
     }
   }
+  if (is_fp8 && !cpuinfo_has_x86_amx_int8()) {
+    // Fall back to ref impl on old platforms because not supported
+    return fp8_qlinear_onednn_ref(
```

Comment:
What's the case of int8 for the old platform? I think it's better that oneDNN can handle it inside.

Reply:
Ok, then let's add a TODO here to remove it in the future.

Reply:
As we synced offline, oneDNN won't handle it themselves. So, I guess there won't be a TODO here. What do you think? Thanks.